doc/development/05_repo_manager.md

```diff
@@ -0,0 +1,202 @@
+so it can be initialized again:
+
+```rust
+/// Clear the publication server. Will fail if it still
+/// has publishers. Or if it does not exist
+pub fn repository_clear(&self) -> KrillResult<()> {
+    self.access.clear()?;
+    self.content.clear()
+}
+```
+
+
+Adding / Removing Publishers
+----------------------------
+
+To add a publisher we have to submit an `rfc8183::PublisherRequest`, and an `Actor`
+so that we attribute the change in the history of the `RepositoryAccess` component.
+The request type can be created from XML supplied by a remote publisher. The 'handle'
+is the local name that we will use for the publisher, and the XML contains a suggestion
+set by the remote party. But, it is possible (and perhaps recommended) to override
+this 'handle' in the request object with something that is unique to *us*, e.g. a UUID.
+
+```rust
+/// Adds a publisher. This will fail if a publisher already exists for the handle in the request.
+pub fn create_publisher(&self, req: rfc8183::PublisherRequest, actor: &Actor) -> KrillResult<()> {
+    let name = req.publisher_handle().clone();
+
+    self.access.add_publisher(req, actor)?;
+    self.content.add_publisher(name)
+}
+```
+
+The publisher is going to need to get RFC8183 Repository Response XML. We can get
+an `rfc8183::RepositoryResponse` which can be printed (`fmt::Display`) as XML:
+
+```rust
+/// Returns the RFC8183 Repository Response for the publisher
+pub fn repository_response(
+    &self,
+    rfc8181_uri: uri::Https,
+    publisher: &PublisherHandle,
+) -> KrillResult<rfc8183::RepositoryResponse> {
+    self.access.repository_response(rfc8181_uri, publisher)
+}
+```
+
+In order to remove a publisher we just need to tell the RepositoryManager its handle (name).
+This removes both the access information (handle, RFC8181 ID cert, ..) and all content for
+the publisher:
+
+```rust
+/// Removes a publisher and all of its content.
+pub fn remove_publisher(&self, name: PublisherHandle, actor: &Actor) -> KrillResult<()> {
+    let publisher = self.access.get_publisher(&name)?;
+    let base_uri = publisher.base_uri();
+
+    self.content.remove_publisher(&name, base_uri, &self.config.repository_retention)?;
+    self.access.remove_publisher(name, actor)
+}
+```
+
+
+Publisher Stats
+---------------
+
+We can get the stats for the complete repository, i.e. with all publishers, as follows:
+
+```rust
+pub fn repo_stats(&self) -> KrillResult<RepoStats> {
+    self.content.stats()
+}
+```
+
+The `RepoStats` structure includes the following information, and can be serialized to
+JSON:
+```rust
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct RepoStats {
+    publishers: HashMap<PublisherHandle, PublisherStats>,
+    session: RrdpSession,
+    serial: u64,
+    last_update: Option<Time>,
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct PublisherStats {
+    objects: usize,
```

Comment:
Presumably usize is a count of objects, not objects themselves. Should this be `num_objects`? (assuming we have the ability to change this at this stage...)